app/lib/user_asset_service.rb

```diff
@@ -0,0 +1,229 @@
+##
+# UserAssetService: manages uploading/localization of "user-provided" static asset files
+# Used primarily for BrandingGroup images, but also supports legacy CKEditor uploads
+##
+
+class UserAssetService
+
+  # GCP Compute project to run reads/writes in
+  COMPUTE_PROJECT = ENV['GOOGLE_CLOUD_PROJECT'].blank? ? '' : ENV['GOOGLE_CLOUD_PROJECT']
+  # Service account JSON credentials
+  SERVICE_ACCOUNT_KEY = !ENV['SERVICE_ACCOUNT_KEY'].blank? ? File.absolute_path(ENV['SERVICE_ACCOUNT_KEY']) : ''
+
+  # Asset Path helpers
+  RAILS_PUBLIC_PATH = Rails.root.join('public', 'single_cell')
+  ASSET_PATHS_BY_TYPE = {
+      ckeditor_pictures: RAILS_PUBLIC_PATH.join('ckeditor_assets', 'pictures'),
+      ckeditor_attachments: RAILS_PUBLIC_PATH.join('ckeditor_assets', 'attachments'),
+      branding_images: RAILS_PUBLIC_PATH.join('branding_groups')
+  }.with_indifferent_access
+  ASSET_TYPES = ASSET_PATHS_BY_TYPE.keys.freeze
+
+  # Bucket info; each Rails environment per project has a bucket
+  STORAGE_BUCKET_NAME = (COMPUTE_PROJECT.dup + "-#{Rails.env}-asset-storage").freeze
+
+  # initialize GCS driver with same credentials as FireCloudClient
+  #
+  # * *yields*
+  #   - +Google::Cloud::Storage+
+  def self.initialize_storage_service
```

Comment:
it seems odd to call this 'initialize' since it's not the one that sets the class variable.  I'd rename this `create_storage_service_instance` or somesuch. (and maybe make it private since the goal is to have access through the storage_service getter)

Reply:
Ah - I see what you mean.  This method is in fact superfluous - `self.storage_service` can call `Google::Cloud::Storage.new` directly.  Good catch.

Reply:
87754ed